airflow/providers/google/cloud/utils/credentials_provider.py

```diff
@@ -241,7 +241,14 @@ def get_credentials_and_project_id(
         project_id = credentials.project_id
 
     if delegate_to:
-        credentials = credentials.with_subject(delegate_to)
+        if hasattr(credentials, 'with_subject'):
+            credentials = credentials.with_subject(delegate_to)
+        else:
+            log.warning(
```

Comment:
I think it would be a better idea to raise an exception to make the user more aware of this issue.

Reply:
Ok. Should this be an `AirflowException`?

Reply:
Yes. It should be AirflowException